### PR TITLE
feat: add AgentRCA dynamic process fault benchmark

### PR DESCRIPTION
**Pull request by EvenSol**
## Summary

Reproduce the normal-only, evidence-grounded core of AgentRCA with a dynamic NeqSim gas/liquid process.

- add an operating-regime-aware normal model for raw process windows
- expose directional mean, variance, range, slope, lag-one, and correlation evidence
- rank auditable physical hypotheses without faulty training data
- generate controlled pressure-sensor bias, export-gas leak, inlet blockage, and imposed multiphase slugging excitation
- retain deterministic JSON evidence for an optional external LLM, without making LLM output authoritative
- document the exact reproduction boundary and safety limitations

Reference: Wei & Fink, *Agentic Root Cause Analysis through Evidence-Grounded Reasoning*, submitted 24 July 2026: https://arxiv.org/abs/2607.22385

## Dynamic benchmark

The public synthetic process uses PR EOS with classic mixing:

- gas: 90 mol% methane / 10 mol% ethane
- liquid: n-decane
- 35 bara, 25 °C feeds
- dynamic 4 m × 1 m horizontal separator
- 20 bara inlet-valve outlet and 5 bara product-valve outlets
- 60 samples at 1 s after warm-up

Faults are introduced progressively. The gas leak is a physical split with an integrated ground-truth mass. Blockage reduces effective valve Kv through the existing fouling model. Sensor bias uses the existing transmitter fault API.

The slug case imposes out-of-phase gas/liquid feed bursts and exercises real separator inventory dynamics. It is explicitly **not** presented as a mechanistic slug-capturing pipeline validation. I avoided building the benchmark on TwoFluidPipe while open issue #2705 questions transient inventory conservation.

## Reproduction boundary

Included from AgentRCA:

- normal-only fitting
- operating-condition matching
- directional statistical evidence
- variance/correlation/temporal evidence
- ranked hypothesis table and trace
- 60 s PRONTO-style window
- human-auditable JSON handoff

Not claimed:

- the unpublished convolutional-autoencoder implementation
- paper-specific LLM prompts/inference
- reproduction of reported PRONTO/TEP accuracy
- exact PRONTO facility dynamics

The paper currently states code will be released after acceptance.

## Validation

Focused tests cover:

- immutable window and schema validation
- nearest-regime matching
- deterministic ranking and JSON traces
- finite positive pressure, non-negative flow, and level bounds
- leak mass and blockage severity
- slugging variance response
- fixed-seed repeated execution
- nearby 0.5 s / 1.0 s time steps
- true scenario in deterministic Top-2 ranking

## Documentation impact

Adds `docs/process/agent-rca-dynamic-fault-benchmark.md` and links it from the process documentation index. The page states the physical assumptions, equations-of-state basis, evidence definitions, open-code limitation, slugging limitation, validation scope, and human-review requirement.

### CODE REVIEW

File: src/main/java/neqsim/process/diagnostics/RcaNormalOperationModel.java [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+
+    List<String> signalNames = normalWindows.get(0).getSignalNames();
+    Map<String, List<RcaProcessWindow>> grouped = new LinkedHashMap<String, List<RcaProcessWindow>>();
+    for (RcaProcessWindow window : normalWindows) {
+      if (window == null) {
+        throw new IllegalArgumentException("normalWindows must not contain null");
+      }
+      if (!signalNames.equals(window.getSignalNames())) {
+        throw new IllegalArgumentException("all normal windows must contain the same ordered signal schema");
+      }
```
[the remark this answers is not in the document]
Comment (EvenSol) [reply src/main/java/neqsim/process/diagnostics/RcaNormalOperationModel.java:58]: Fixed. `fit(...)` now requires every normal window to expose the same operating-condition key set, and a regression test covers mismatched dimensions. Regime distances are therefore always evaluated in one coordinate space.

File: src/main/java/neqsim/process/diagnostics/RcaProcessWindow.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      double[] values = entry.getValue();
+      if (values == null || values.length < 3) {
+        throw new IllegalArgumentException("signal " + name + " must contain at least three samples");
+      }
+      if (expectedSamples < 0) {
+        expectedSamples = values.length;
+      } else if (values.length != expectedSamples) {
+        throw new IllegalArgumentException("all signals must contain the same number of samples");
+      }
+      double[] copy = values.clone();
```
[the remark this answers is not in the document]
Comment (EvenSol) [reply src/main/java/neqsim/process/diagnostics/RcaProcessWindow.java]: Fixed. The builder remains the single defensive-copy boundary; the constructor validates and adopts that private array instead of cloning it a second time. The existing input-mutation test still verifies isolation.
